tests/expectations/test_condition_validators.py

```diff
@@ -0,0 +1,137 @@
+"""Tests for condition validators on row_condition in Expectations."""
+
+import pytest
+
+from great_expectations.expectations.conditions import Column
+from great_expectations.expectations.core import (
+    ExpectColumnValuesToBeInSet,
+    ExpectTableRowCountToEqual,
+)
+
+pytestmark = pytest.mark.unit
+
+
+class TestAndConditionValidator:
+    """Tests for AndCondition validator when passed to Expectation."""
+
+    def test_flatten_nested_and_conditions(self):
+        """Test that nested AndConditions are flattened when passed to Expectation."""
+        column_1 = Column(name="column_1")
+        column_2 = Column(name="column_2")
+        column_3 = Column(name="column_3")
+
+        row_condition = (column_1 < 8) & ((column_2 > 8) & (column_3 == 8))
+
+        # Pass to an Expectation - should flatten
+        expectation = ExpectColumnValuesToBeInSet(
+            column="test_column", value_set=["a", "b"], row_condition=row_condition
+        )
+
+        # Verify flattening occurred - should have 3 conditions instead of nested structure
+        assert len(expectation.row_condition.conditions) == 3
+
+    def test_error_on_or_within_and(self):
+        """Test that OrConditions nested within AndConditions raise error in Expectation."""
+        column_1 = Column(name="column_1")
+        column_2 = Column(name="column_2")
+        column_3 = Column(name="column_3")
+
+        # OR within AND
+        row_condition = (column_1 < 8) & ((column_2 > 8) | (column_3 == 8))
+
+        with pytest.raises(ValueError, match="AND groups cannot contain OR conditions"):
+            ExpectColumnValuesToBeInSet(
+                column="test_column", value_set=["a", "b"], row_condition=row_condition
+            )
+
+
+class TestOrConditionValidator:
+    """Tests for OrCondition validator when passed to Expectation."""
+
+    def test_error_on_nested_or_within_or_within_and(self):
+        """Test that nested OR structures are caught by validation."""
+        column_1 = Column(name="column_1")
+        column_2 = Column(name="column_2")
+        column_3 = Column(name="column_3")
+
+        nested_or = (column_1 > 8) | ((column_2 == 8) | (column_3 == 9))
+
+        with pytest.raises(ValueError, match="OR groups cannot contain nested OR conditions"):
```

Comment:
Nit: Another wording suggestion.
```suggestion
        with pytest.raises(ValueError, match="Multiple OR groups cannot be combined"):
```

Reply:
The problem with the configuration is that there is an OR condition at a lower level than an OR condition. That's why we are saying "contain". If the condition was `(column_1 < 8) | (column_2 > 8) | (column_3 == 8)` it would be valid.